version: "3.7"

services:

  pi-hole:
    image: pihole/pihole:2025.07.1@sha256:f84c1654bfbafc44e2ac8447a1fa2fb739eae66395b4c179d2c59cb93e1321f3
    hostname: pi-hole
    restart: on-failure
    environment:
      - TZ=UTC
      - VIRTUAL_HOST=${APP_DOMAIN}
    ports:
      - "53:53/tcp"
      - "53:53/udp"
      - "8082:80/tcp"
    volumes:
      - ${APP_DATA_DIR}/pihole/etc/pihole:/etc/pihole
      - ${APP_DATA_DIR}/pihole/etc/dnsmasq.d:/etc/dnsmasq.d
    depends_on:
      - dns-crypt
    networks:
      pi-hole-dns-crypt:
        ipv4_address: 172.20.0.2

  dns-crypt:
    image: zydou/dnscrypt:2025-07-18@sha256:a2226d762a5b74c20097661dc8443d9145ae4285c2e792feb3fd2bb011c9be9b
    hostname: dns-crypt
    restart: on-failure
    environment:
      - TZ=UTC
    volumes:
      - ${APP_DATA_DIR}/dnscrypt/dnscrypt-proxy.toml:/etc/dnscrypt-proxy/dnscrypt-proxy.toml
    dns:
      - "1.1.1.1"
      - "1.0.0.1"
    networks:
      pi-hole-dns-crypt:
        ipv4_address: 172.20.0.3

networks:
  pi-hole-dns-crypt:
    driver: bridge
    ipam:
      config:
        - subnet: 172.20.0.0/29
